### ISSUE TITLE
Compression will pause if (`app is in background` && `Screen is locked`) 

### ISSUE THREAD

**Issue by symphonyrecords** [brarcher/video-transcoder#138]
I've tried  `wavelock` ,  `AlarmManager` with no success. I don't know maybe I'm doing something wrong but still compression process will pause when app goes in background and screen locks.
Any solutions!?

**Comment by brarcher** [comment]
Hm. I attempted to set it up so that the service which does the encoding runs in the foreground, so that it does get CPU when it needs. I though that would also hold a power lock so it could still encode when the screen is off. The background service would run even if the app was not visible. I'll have to see if I'm wrong about the power lock.

What version of Android are you running on?

**Comment by symphonyrecords** [comment]
Hi, thank you for your respond. I saw this issue in `Honor 8 Lite - API24`. But in `Samsung j5 Prime API24` it works as it's expected.

**Comment by brarcher** [comment]
Hm. The device I'm testing with is also Android 7, and it appeared to work there as well. What I do see is that the encoding progress is slower when the screen is off. It may be that the CPU is being throttled in my case when the screen is off.

Are you able to capture logs when this happens? That may help in figuring out what is happening.

**Comment by licaon-kter** [comment]
@symphonyrecords You did disable battery optimizations for the app?

You did add it to whatever "protected apps" whitelist you can use in Android Setting, right?

**Comment by symphonyrecords** [comment]
@brarcher , Yes, when app is in background the process gets slower, and in about 20 sec it will completely pause until I unlock the screen. Yes I figured this issue while I was capturing the log, also the log gets paused when this happens. I Already tried many things I'm trying to figure this out for about a week now.

**Comment by symphonyrecords** [comment]
@licaon-kter  I allow app to ignore battery optimizations. didn't worked

**Comment by symphonyrecords** [comment]
There is another note that might worth mentioning.
In android studio when app is attached to adb through `USB Cable` everything works fine. 
This issue only appears if I connect device to adb through `Wifi`.

**Comment by brarcher** [comment]
I tried an experiment yesterday, setting the app to encode the audio from a 4 hour video overnight while plugged in. 8 hours later it was ~90% complete. After another ~20 minutes it completed.

From this, on my device (LineageOS running Android 7.1.1) I see that when the screen is off and adb is disconnected the CPU is being throttled but not being completely stopped.

The service in the app which monitors ffmpeg is run as a foreground service. Looking up the docs on that Android does not limit the execution of such services. (If it was a background service it could stop execution after a time when the screen is off).

There is a new power lock hint that can be used in Android 7 that requests the device provide sustained performance: 
https://source.android.com/devices/tech/power/performance

I do not know if that could help the issue or not while encoding. It is worth looking into.

**Comment by symphonyrecords** [comment]
Thank you. I appreciate it, I'll look into it.

**Comment by symphonyrecords** [comment]
@brarcher  This is not related to this issue but I have [this](https://github.com/symphonyrecords/FFmpegUtils) class that will get log from ffmpeg output and parse necessary metadata. So it will help you to remove `ffprobe` from your app and reduce app size . I've tested on a few media files and it's working fine so far.

**Comment by symphonyrecords** [comment]
@brarcher fair enough
